<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="utf-8">
		<title>Throwing custom exception types in JavaScript | </title>
		
		<link rel="stylesheet" href="../../../../../tore/@/fonts/listΣ.css?v=1">
<link rel="stylesheet" href="../../../../../tore/layd/deff/&/aefb44b7dac371a3860e9079298a950e.ructΣ.css">
<link rel="stylesheet" href="../../../../../tore/&/2290bf7aeadc4261337d6f570a3cf0c5.apprΣ.css">

<script src="../../../../../tore/&/77748427269d1c81490f5acb6ce46fc1.apprΣ.js" defer></script>


<meta name="yandex-verification" content="00eb45b7559e5c68">
<meta name="google-site-verification" content="Af0TM2OrwIEtjoQXxYz2-1C-21E1m7DB23_kWpE9sLg">
<meta name="viewport" content="width=device-width, initial-scale=1.0">		
		<meta name="description" content="In this lesson, we will learn how to throw custom types of exceptions in JavaScript.">
		<link rel="shortcut icon" type="image/x-icon" href="../../../../../favicon.ico">
	</head>
	<body id="top" data-gq="D">
		<div data-r="ruct/wrap">
			<header>
				<div data-r="ruct/topp">
	<div data-r="topp/left">
		<div data-w="nict/logo">
	<img data-e="img" data-x src="" alt="">
</div>	</div>
	<div data-r="topp/righ">
		offline version v3	</div>
	<div data-r="topp/cent">
		<br><br>	</div>
</div>								
				<div data-r="ruct/navi">
					<div data-w="nict/over">
			<span data-e="hash" data-x>⊗jsSpExcThrUT</span>
				<span data-e="lofo" data-x>147 of 281</span>
				<a href="../../../../../en/javascript/book/supreme.html" data-e="link" data-x>menu</a>
	</div>					<nav data-w="nict/mainmenu">
	<section data-e="main" data-x data-gm-acte="T">
		<span data-q-href="/en/markup/" data-e="open" data-x data-m-targ="markup">Markup</span>
		<span data-q-href="/en/javascript/" data-e="open" data-x data-m-targ="javascript">JavaScript</span>
	</section>
	
	<section data-e="subb" data-x data-gm-acte="F" data-m-tabb="markup">
		<a href="../../../../../en/markup/manual/html.html">HTML manual</a>
		<a href="../../../../../en/markup/manual/css.html">CSS manual</a>
	</section>
	
	<section data-e="subb" data-x data-gm-acte="F" data-m-tabb="javascript">
		<a href="../../../../../en/javascript/book/prime.html">Prime book</a>
		<a href="../../../../../en/javascript/book/practice.html">Practice book</a>
		<a href="../../../../../en/javascript/book/supreme.html">Supreme book</a>
		<a href="../../../../../en/javascript/manual.html">Manual</a>
	</section>
</nav>									</div>
			</header>
			<main>
				<div data-w="nict/hett">
	<h1>Throwing custom exception types in JavaScript</h1>
</div>				<div data-p="cont/pane">
<p>
	The number of built-in in JavaScript 
	exceptions is not that large, and they 
	most often cannot satisfy all our needs 
	for different types of exceptions. Therefore, 
	JavaScript has the built-in ability to 
	create exceptions with a custom type.
</p>
<p>
	There are different ways to do this. The 
	simplest one is to pass an object with 
	the keys <code data-gs="comd/c" data-x>name</code> and <code data-gs="comd/c" data-x>message</code> 
	to <code data-gs="comd/c" data-x>throw</code>:
</p>
<div data-p="bock/pane">
	<code data-w="bock/code" data-lo="high" data-m-lang="javascript">try {
	throw {name: 'MyError', message: 'an exception text'};
} catch (error) {
	console.log(error.name); // 'MyError'
	console.log(error.message); // 'an exception text'
}</code>
</div>
<div data-w="cont/task">
	<p>
		Previously, we made a function that 
		throws an exception when dividing 
		by zero:
	</p>
	<div data-p="bock/pane">
		<code data-w="bock/code" data-lo="high" data-m-lang="javascript">function div(a, b) {
	if (b !== 0) {
		return a / b;
	} else {
		throw new Error('division by zero error');
	}
}</code>
	</div>
	<p>
		Modify this function so that it throws 
		an exception with some kind of type we 
		invented, for example, 
		<code data-gs="comd/c" data-x>DivisionByZeroError</code>.
	</p>
</div>
<div data-w="cont/task">
	<p>
		Previously, you made a function that 
		throws an exception when trying to 
		extract the square root of a negative 
		number. Modify your function so that 
		it throws an exception with the type 
		you invented. Think carefully about 
		the name of the exception so that 
		this name is successful.
	</p>
</div>
</div>			</main>
				<div data-w="nict/nebr">
					<a href="../../../../../en/javascript/book/supreme/exceptions/throwing-types.html" data-e="prev" data-x>
				<span data-e="arow" data-x>&larr;</span>
			</a>
							<a href="../../../../../en/javascript/book/supreme/exceptions/data-attr-example.html" data-e="next" data-x>
				<span data-e="arow" data-x>&rarr;</span>
			</a>
			</div>
			<footer data-w="ruct/foot">
				<div data-w="nict/finf">
	<span data-e="text" data-x>
		Trepachev Dmitry &copy; 2012-2024	</span>
	<br>
	<a data-e="link" data-x href="https://t.me/trepachev_dmitry">t.me/trepachev_dmitry</a>
</div>							</footer>
			<div data-w="nict/jump">
	<a href="#top"><span>&uarr;</span></a>
</div>			<div data-w="nict/lagr">
	<a href="../../../../../en/javascript/book/supreme/exceptions/throwing-user-types.html" data-m-curr="T">en</a><a href="../../../../../ru/javascript/book/supreme/exceptions/throwing-user-types.html" data-m-curr="F">ru</a></div>					</div>
		
		<script src="../../../../../tore/@/metrikaΣ.js?v=1"></script>	</body>
</html>